app/controllers/DeveloperForm.scala

```
@@ -44,6 +44,32 @@ class DeveloperForm(override val controllerComponents: ControllerComponents, dyn
     createKeyForm.bindFromRequest.fold[Future[Result]](handleInvalidForm, handleValidForm)
   }
 
+  def deleteKeys(id: String, hash: String) = Action.async { implicit request =>
+    if (md5(id) != hash)
+      Future.successful(Forbidden)
+    else
+      logic.deleteKeys(id).recover {
+        case err =>
+          awsEmail.sendEmailDeletionFailed(id, err)
+          ()
```

Should we return an appropriate error message here?

---

I'm on the fence. The time-sensitive nature of the key cleanup process means if any error occurs, we should react immediately. _If_ an error message was displayed to the user, say "Something's wrong, please retry later", there is a likelihood they will forget and lose their key.